keyboards/evyd13/atom47/keymaps/phsc138/keymap.c

```diff
@@ -0,0 +1,390 @@
+        //key has not been interrupted, but they key is still held. Means you want to send a 'HOLD'.
+        else return SINGLE_HOLD;
+    } else if(state->count == 2) {
+        /*
+         * DOUBLE_SINGLE_TAP is to distinguish between typing "pepper", and actually wanting a double tap
+         * action when hitting 'pp'. Suggested use case for this return value is when you want to send two
+         * keystrokes of the key, and not the 'double tap' action/macro.
+        */
+        if(state->interrupted) return DOUBLE_SINGLE_TAP;
+        else if(state->pressed) return DOUBLE_HOLD;
+        else return DOUBLE_TAP;
+    }
+    //Assumes no one is trying to type the same letter three times (at least not quickly).
+    //If your tap dance key is 'KC_W', and you want to type "www." quickly - then you will need to add
+    //an exception here to return a 'TRIPLE_SINGLE_TAP', and define that enum just like 'DOUBLE_SINGLE_TAP'
+    if(state->count == 3) {
+        if(state->interrupted || !state->pressed) return TRIPLE_TAP;
+        else return TRIPLE_HOLD;
+    }
+    else return 8; //magic number. At some point this method will expand to work for more presses
+}
+
+//instanalize an instance of 'tap' for the 'pn' tap dance.
+static tap pn_tap_state = {
+    .is_press_action = 1,
+    .state = 0
+};
+
+void pn_finished(qk_tap_dance_state_t *state, void *user_data) {
+    pn_tap_state.state = cur_dance(state);
+    switch(pn_tap_state.state) {
+        case SINGLE_TAP: layer_on(_PROG); break;
+        case SINGLE_HOLD: layer_on(_NUM); break;
+        //case DOUBLE_TAP: register_code(KC_ESC); break;
+        //case DOUBLE_HOLD: register_code(KC_LALT); break;
+        //case DOUBLE_SINGLE_TAP: register_code(KC_X); unregister_code(KC_X); register_code(KC_X);
+        //Last case is for fast typing. Assuming your key is `f`:
+        //For example, when typing the word `buffer`, and you want to make sure that you send `ff` and not `Esc`.
+        //In order to type `ff` when typing fast, the next character will have to be hit within the `TAPPING_TERM`, which by default is 200ms.
+    }
+}
+
+void pn_reset(qk_tap_dance_state_t *state, void *user_data) {
+    switch(pn_tap_state.state) {
+        //case SINGLE_TAP: unregister_code(KC_X); break;
+        case SINGLE_HOLD: layer_off(_NUM); break;
+        //case DOUBLE_TAP: unregister_code(KC_ESC); break;
+        //case DOUBLE_HOLD: unregister_code(KC_LALT);
+        //case DOUBLE_SINGLE_TAP: unregister_code(KC_X);
+    }
+    pn_tap_state.state = 0;
+}
+
+static tap lalt_tap_state = {
+    .is_press_action = 1,
+    .state = 0
+};
+
+void lalt_finished(qk_tap_dance_state_t *state, void *user_data) {
+    lalt_tap_state.state = cur_dance(state);
+    switch(lalt_tap_state.state) {
+        case SINGLE_TAP: register_code(KC_LBRC); break;
+        case SINGLE_HOLD: register_code(KC_LALT); break;
+    }
+}
+
+void lalt_reset(qk_tap_dance_state_t *state, void *user_data) {
```

Comment:
```suggestion
void lalt_reset(tap_dance_state_t *state, void *user_data) {
```

Reply:
Fixed this comment here: [d9e04e8](https://github.com/qmk/qmk_firmware/pull/18768/commits/d9e04e80af7200159a9307be8ed3559a94052311)